src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -675,6 +712,236 @@ GroupPrincipal getGroupPrincipal(Member group) {
     }
   }
 
+  @Override
+  public void getModifiedDocIds(DocIdPusher pusher)
+      throws IOException, InterruptedException {
+    // Log in using CWS and use that token for the search.
+    Authentication authentication = this.soapFactory.newAuthentication();
+    String authenticationToken =
+        getAuthenticationToken(this.username, this.password);
+
+    ArrayList<DocId> docIds = new ArrayList<DocId>();
+    int resultCount = 0;
+    do {
+      String query = getLastModifiedQuery();
+      log.log(Level.FINER, "getModifiedDocIds query: {0}", query);
+      Element documentElement =
+          getXmlSearchResults(authenticationToken, query);
+      resultCount = getXmlSearchCount(documentElement);
+      log.log(Level.FINER, "getModifiedDocIds result count: " + resultCount);
+      if (resultCount > 0) {
+        NodeList searchResults =
+            documentElement.getElementsByTagName("SearchResult");
+        for (int i = 0; i < searchResults.getLength(); i++) {
+          String docId = getXmlSearchDocId((Element) searchResults.item(i));
+          if (docId != null) {
+            docIds.add(new DocId(docId));
+          }
+        }
+        // Cache the last object's id/date/time
+        if (docIds.size() > 0) {
+          Long nodeId = Longs.tryParse(
+              docIds.get(docIds.size() - 1).getUniqueId().split(":")[1]);
+          DocumentManagement documentManagement =
+              this.soapFactory.newDocumentManagement(authenticationToken);
+          Node node = getNodeById(documentManagement, nodeId);
+          if (node == null) {
+            this.lastModDocId = null;
+            this.lastModDate = null;
+            this.lastModTime = null;
+          } else {
+            XMLGregorianCalendar xmlCalendar = node.getModifyDate();
+            if (xmlCalendar != null) {
+              Date date = xmlCalendar.toGregorianCalendar().getTime();
+              this.lastModDocId = nodeId;
+              this.lastModDate = new SimpleDateFormat("yyyyMMdd").format(date);
+              this.lastModTime = new SimpleDateFormat("HHmmss").format(date);
+            }
+          }
+        }
+      }
+    } while (resultCount > 0);
+    log.log(Level.FINER, "Sending modified doc ids: {0}", docIds);
```

Comment:
Smaller log messages. 

To do that, we'd either log each DocId individually or add code to track which DocIds belong to a loop iteration just for the purpose of creating the log message.